### PR TITLE
MSEARCH-42: [PT3] Implement inventory-reindex endpoint.

### PR DESCRIPTION
**Pull request by bohdan-suprun**

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -141,6 +157,11 @@
       "permissionName": "search.config.languages.item.delete",
       "displayName": "Search - removes a supported language by id",
       "description": "Removes a supported language by id"
+    },
+    {
+      "permissionName": "search.index.inventory.reindex.post",
+      "displayName": "Search - starts a reindex operation",
+      "description": "Starts a reindex operation"
```
Comment (hjiebsco) [inline descriptors/ModuleDescriptor-template.json:164]: Not consistent. `Inventory` is not mentioned in display name and description.

File: src/main/java/org/folio/search/service/IndexService.java [hunk 2]
```diff
@@ -92,6 +95,12 @@ public void createIndexIfNotExist(String resourceName, String tenantId) {
     var index = getElasticsearchIndexName(resourceName, tenantId);
     if (!indexRepository.indexExists(index)) {
       createIndex(resourceName, tenantId);
+    } else if (log.isInfoEnabled()) {
+      log.info("Index already exists [index={}]", index);
```
Comment (hjiebsco) [inline src/main/java/org/folio/search/service/IndexService.java:99]: Is this needed? Method name is `createIndexIfNotExist`

File: src/test/java/org/folio/search/service/IndexServiceTest.java [hunk 3]
```diff
@@ -105,4 +107,11 @@ void createIndexIfNotExist_shouldNotCreateIndex_alreadyExist() {
 
     verify(indexRepository, times(0)).createIndex(eq(indexName), any(), any());
   }
+
+  @Test
+  void reindexInventory_positive() {
```
Comment (hjiebsco) [inline src/test/java/org/folio/search/service/IndexServiceTest.java:112]: Looks like it is a pattern to use `_`. Fine by me.

**Review by hjiebsco — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (bohdan-suprun) [reply descriptors/ModuleDescriptor-template.json]: Thanks, updated.

File: src/main/java/org/folio/search/service/IndexService.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/search/service/IndexService.java]: The log statement is removed.

**Review by pfilippovEPAM — APPROVED** [review]

**Review by MikhailFokanov — APPROVED** [review]

**Pull request merged**